test/unit_test/deepdoc/vision/test_tsr_row_merge.py

```
@@ -0,0 +1,306 @@
+
+def _bug_18569_row_components():
+    """The 13 ``table row`` components reported in #18569, page 11.
+
+    Heights and y-positions taken verbatim from the issue. The first row
+    (7.1pt) is the split-line / degenerate detection that the fix drops
+    outright; the rest are the wrapped-paragraph lines that the fix merges
+    into a single component.
+    """
+    y0s = [7980.1, 7989.6, 8009.1, 8026.6, 8045.8, 8064.9, 8082.4, 8134.3, 8151.8, 8169.0, 8186.5, 8203.3, 8220.0]
+    heights = [7.1, 19.5, 17.2, 19.3, 19.2, 17.4, 51.6, 17.6, 17.5, 17.5, 17.0, 17.6, 28.7]
+    return [_row("table row", y, y + h) for y, h in zip(y0s, heights)]
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)